src/coreml.cs

```diff
@@ -456,25 +457,27 @@ interface MLBatchProvider {
 	}
 
 	[Watch (5,0), TV (12,0), Mac (10,14, onlyOn64: true), iOS (12,0)]
-	[Protocol]
+	[BaseType (typeof(NSObject))]
+	[Protocol, Model]
```

Comment:
nice workaround (does not always work but it's good here)
@rolfbjarne I suspect that might run into naming conflict in the future, was your fix only in master (or 15.8) ?

Reply:
oh nice catch @spouliot, @VincentDondain  if `[Model (AutoGeneratedName = true)]` is not available please add it here https://github.com/xamarin/xamarin-macios/issues/4189 so we do not forget.

Reply:
The AutoGeneratedName fix is in d15-7 as well.